onnxruntime/core/providers/webnn/builders/impl/squeeze_unsqueeze_op_builder.cc

```diff
@@ -78,15 +78,25 @@ Status SqueezeUnsqueezeOpBuilder::AddToModelBuilderImpl(ModelBuilder& model_buil
     }
   }
 
-  if (axes_data.size() > 0) {
-    options.set("axes", emscripten::val::array(axes_data));
-  }
-
   emscripten::val output = emscripten::val::undefined();
   if (op_type == "Squeeze") {
+    if (axes_data.size() > 0) {
+      options.set("axes", emscripten::val::array(axes_data));
+    }
     output = model_builder.GetBuilder().call<emscripten::val>("squeeze", input, options);
   } else if (op_type == "Unsqueeze") {
-    output = model_builder.GetBuilder().call<emscripten::val>("unsqueeze", input, options);
+    // Use WebNN's reshape to implement Unsqueeze.
+    std::vector<int32_t> new_shape;
+    std::transform(
+        input_shape.begin(), input_shape.end(), std::back_inserter(new_shape),
+        [](int64_t data) -> int32_t { return SafeInt<int32_t>(data); });
+    // Sort axes_data in ascending order.
+    std::sort(axes_data.begin(), axes_data.end());
+    // Expand new_shape according to axes_data.
+    for (const int32_t& axis : axes_data) {
+      new_shape.insert(new_shape.begin() + axis, 1);
```

Comment:
Do we handle negative axes?

In the DML EP, there is this line `HandleNegativeAxes`, which also validates the axis range:

```c++
    void UnsqueezeHelper::Initialize(
        IKernelInformationAdapter const& kernelInformation,
        IShapeInformationAdapter const& shapeInformation,
        uint32_t opsetVersion
        )
    {
        if (opsetVersion >= 13) // Axes are a dynamic input parameter.
        {
            ReadCpuLocalTensorIntoInt32(kernelInformation.GetConstantInputTensor(1), /*out*/ m_axes);
        }
        else // Axes were a constant attribute parameter.
        {
            m_axes = kernelInformation.GetAttributes().GetOptionalAttributeVectorInt32(AttrName::Axes);
        }
        std::vector<DimensionType> inputDimensions = shapeInformation.GetInputTensorShape(0);

        const uint32_t outputDimensionCount = gsl::narrow_cast<uint32_t>(inputDimensions.size() + m_axes.size());
        HandleNegativeAxes(/*inout*/ m_axes, outputDimensionCount);
        std::sort(m_axes.begin(), m_axes.end());
    }
```

Reply:
Handled. See above comment.